initialize `env_logger`

ozkanonur opened the pull request:
This is for quick/temp fix. Will need to refactor and exclude our custom log implementation.

cc @Alrighttt

ozkanonur commented:
I am not sure if wasm side will be as stable as before. @yurii-khi could be nice if you can test/check this for wasm

yurii-khi commented:
> I am not sure if wasm side will be as stable as before. @yurii-khi could be nice if you can test/check this for wasm

Hey! Do you have any specific test-cases in mind, or just general smoke testing?

ozkanonur commented:
> > I am not sure if wasm side will be as stable as before. @yurii-khi could be nice if you can test/check this for wasm
> 
> Hey! Do you have any specific test-cases in mind, or just general smoke testing?

Just log specific testing for wasm

yurii-khi commented:
> > > I am not sure if wasm side will be as stable as before. @yurii-khi could be nice if you can test/check this for wasm
> > 
> > 
> > Hey! Do you have any specific test-cases in mind, or just general smoke testing?
> 
> Just log specific testing for wasm. I want to make sure I didn't break any of the log functionality for wasm

Right now it breaks our logger, most probably because of missing `LogLevel`, @anarkiVan could you please take a closer look?

ozkanonur commented:
Because `LogLevel` is excluded from mm2. So it shouldn't be provided. But I am also not sure if `env_logger` will work for wasm. Let me keep it simple and rollback `LogLevel` for wasm target. Not worth making breaking change there.

yurii-khi commented:
> 

I think best way would be to use exact js file we're using on webdex side:
https://github.com/KomodoPlatform/WebDEX/blob/dev/web/src/index.js

But I'd recommend to install flutter and run app locally, it's very simple and will provide reliable results.
https://github.com/KomodoPlatform/WebDEX/blob/add/docs/docs/project_setup.md
https://github.com/KomodoPlatform/WebDEX/blob/add/docs/docs/update_wasm_module.md

ozkanonur commented:
We removed the custom `LogLevel` implementation completely. Since it increases the complexicty for literally "no reason". It doesn't provide any flexibility or feature.

What happens if I remove it for wasm as well? I removed them in this PR and need a feedback from gui team. Does `LogLevel` thing is really needed for gui side?

For testing it, https://github.com/KomodoPlatform/WebDEX/blob/b8f05db785783889e659ea67f36dd7313f76d55c/web/src/index.js#L37 this line needs to be removed with `LogLevel` all together because they don't exists anymore with this PR.


cc @yurii-khi @anarkiVan

yurii-khi commented:
> We removed the custom `LogLevel` implementation completely. Since it increases the complexicty for literally "no reason". It doesn't provide any flexibility or feature.
> 
> What happens if I remove it for wasm as well? I removed them in this PR and need a feedback from gui team. Does `LogLevel` thing is really needed for gui side?
> 
> For testing it, https://github.com/KomodoPlatform/WebDEX/blob/b8f05db785783889e659ea67f36dd7313f76d55c/web/src/index.js#L37 this line needs to be removed with `LogLevel` all together because they don't exists anymore with this PR.
> 
> cc @yurii-khi @anarkiVan

I think we can get rid of it on our side as well, just need to wait until changes propagate to dev branch.

ozkanonur commented:
It can be nice if you can test this branch on gui side for wasm, to see if there is nothing wrong after removing `LogLevel`

yurii-khi commented:
> It can be nice if you can test this branch on gui side for wasm, to see if there is nothing wrong after removing `LogLevel`

Tested `ed5c72b3b` build with changes on our side, mentioned above (deleted `LogLevel` completely), and if works fine, was able to run the app and swap some RICK->MORTY. I have a few questions:
1. Does it pass all the mm2 logs to `handle_log` atm? It looks like amount of log output has been reduced significantly.
2. Is there a way to separate errors from other messages, so we can treat them differently?

shamardy commented on CHANGELOG.md:
```diff
@@ -20,6 +20,7 @@
 - `mm2_git` crate was added to provide an abstraction layer on Git for doing query/parse operations over the repositories [#1636](https://github.com/KomodoPlatform/atomicDEX-API/pull/1636)
 
 **Enhancements/Fixes:**
+- Use `env_logger` to achieve flexible log filtering
```
Please add the PR tag here :)

shamardy reviewed (approved):
🔥
